src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,325 @@
+ *      3) an `IndexRegistry` that keeps track of an ordered list of operators for each quorum
+ * 
+ * @author Layr Labs, Inc.
+ */
+contract BLSRegistryCoordinatorWithIndices is Initializable, IBLSRegistryCoordinatorWithIndices {
+    using BN254 for BN254.G1Point;
+
+    /// @notice the EigenLayer Slasher
+    ISlasher public immutable slasher;
+    /// @notice the Service Manager for the service that this contract is coordinating
+    IServiceManager public immutable serviceManager;
+    /// @notice the BLS Pubkey Registry contract that will keep track of operators' BLS public keys
+    IBLSPubkeyRegistry public immutable blsPubkeyRegistry;
+    /// @notice the Stake Registry contract that will keep track of operators' stakes
+    IStakeRegistry public immutable stakeRegistry;
+    /// @notice the Index Registry contract that will keep track of operators' indexes
+    IIndexRegistry public immutable indexRegistry;
+    /// @notice the mapping from quorum number to the maximum number of operators that can be registered for that quorum
+    mapping(uint8 => OperatorSetParam) public quorumOperatorSetParams;
+    /// @notice the mapping from operator's operatorId to the updates of the bitmap of quorums they are registered for
+    mapping(bytes32 => QuorumBitmapUpdate[]) internal _operatorIdToQuorumBitmapHistory;
+    /// @notice the mapping from operator's address to the operator struct
+    mapping(address => Operator) internal _operators;
+    /// @notice the dynamic-length array of the registries this coordinator is coordinating
+    address[] public registries;
+
+    modifier onlyServiceManagerOwner {
+        require(msg.sender == serviceManager.owner(), "BLSIndexRegistryCoordinator.onlyServiceManagerOwner: caller is not the service manager owner");
+        _;
+    }
+
+    constructor(
+        ISlasher _slasher,
+        IServiceManager _serviceManager,
+        IStakeRegistry _stakeRegistry,
+        IBLSPubkeyRegistry _blsPubkeyRegistry,
+        IIndexRegistry _indexRegistry
+    ) {
+        slasher = _slasher;
+        serviceManager = _serviceManager;
+        stakeRegistry = _stakeRegistry;
+        blsPubkeyRegistry = _blsPubkeyRegistry;
+        indexRegistry = _indexRegistry;
+    }
+
+    function initialize(OperatorSetParam[] memory _operatorSetParams) external initializer {
+        // the stake registry is this contract itself
+        registries.push(address(stakeRegistry));
+        registries.push(address(blsPubkeyRegistry));
+        registries.push(address(indexRegistry));
+
+        // set the operator set params
+        require(IVoteWeigher(address(stakeRegistry)).quorumCount() == _operatorSetParams.length, "BLSIndexRegistryCoordinator: operator set params length mismatch");
+        for (uint8 i = 0; i < _operatorSetParams.length; i++) {
+            quorumOperatorSetParams[i] = _operatorSetParams[i];
```

Comment:
probably want an internal setter function for this which checks conditions (e.g. valid percentages) and emits an event.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/4f862c59d143ff36a451a93de69861e89567b1ca